chore(client): Merge client and engines-core packages

SevInf opened the pull request:
Moves most of `engine-core` package into `core/runtime/engine` subdirectory of the client.
Moves some utitlites from the `engine-core` into various `internals` sub modules.

This allows us to use `internals` helpers within what used to be `engines-core` code. In the future, it would also allow us to simplify the code by reusing some of the client-side abstractions.

**Note for reviewers**: It is mostly files moving and imports updating, whatever deserves some attention I commented on.

Fix prisma/client-planning#358
Unblock prisma/client-planning#340

SevInf commented on packages/client/src/runtime/core/engines/common/utils/getErrorMessageWithLink.test.ts:
```diff
@@ -0,0 +1,31 @@
+
+import { getErrorMessageWithLink } from './getErrorMessageWithLink'
+
+test('basic serialization', () => {
```
Split from `errors.tests.ts` into it's own test

SevInf commented on packages/client/src/runtime/core/engines/common/utils/util.ts:
```diff
@@ -0,0 +1,28 @@
+import crypto from 'crypto'
+import newGitHubIssueUrl from 'new-github-issue-url'
+
+export function getGitHubIssueUrl({
```
Most of the helpers from here have moved to `internals`, I guess that's why github thinks it is a new file.

SevInf commented on packages/client/src/runtime/core/engines/library/ExitHooks.ts:
```diff
@@ -15,11 +16,12 @@ export class ExitHooks {
-    this.installHook('SIGINT', true)
-    this.installHook('SIGUSR2', true)
-    this.installHook('SIGTERM', true)
+    process.once('exit', () => this.triggerAllListeners())
```
Not needed after #18848 is merged, TS just forced me to fix the same issue as well

SevInf commented on packages/client/src/runtime/core/engines/library/LibraryEngine.test.ts:
```diff
@@ -90,7 +91,12 @@ jest.mock('fs', () => {
+      if (fileName === '/mock/schema.prisma') {
+        return ''
+      }
+      return original.readFileSync(fileName, ...rest)
```
Fallback to original is now needed, because non-treeshaked `internals` will import `prisma-ft-wasm`, which will try to read wasm file, which will if fail if `readFileSync` is mocked uncondinitonally.

SevInf commented on packages/internals/src/engine-commands/getEngineVersion.ts:
```diff
@@ -8,6 +7,10 @@ import * as TE from 'fp-ts/TaskEither'
 import { resolveBinary } from '../resolveBinary'
 import { loadLibrary } from '../utils/load'
 
+type NodeAPILibrary = {
```
We could've moved whole library type definition to internals, but since it is just one function, I've decided it is not worth it.

SevInf commented on packages/internals/src/utils/plusX.ts:
```diff
@@ -0,0 +1,15 @@
+import Debug from '@prisma/debug'
```
Not a new code, moved from `engine-core/common/utils`

SevInf commented on .github/workflows/test.yml:
```diff
@@ -983,20 +983,6 @@ jobs:
           flags: debug,${{ matrix.os }},library,binary
           name: debug-${{ matrix.os }}
 
-      - run: pnpm run test
```
Removing the jobs from GH Actions

SevInf commented on packages/client/src/runtime/core/engines/data-proxy/errors/utils/responseToError.test.ts:
```diff
@@ -141,35 +137,3 @@ describe('responseToError', () => {
   })
 })
-
-describe('getErrorMessageWithLink', () => {
```
Moved to separate file

SevInf commented on packages/client/src/runtime/core/engines/data-proxy/errors/utils/responseToError.test.ts:
```diff
@@ -1,13 +1,9 @@
 const response = (body: string, code?: number, requestId?: string): RequestResponse => ({
   json: () => Promise.resolve(body),
-  text: () => body,
+  text: () => Promise.resolve(body),
```
This method had incorrect return type, but since `engines-core` tests were not typechecked it was not caught.

SevInf commented on packages/internals/package.json:
```diff
@@ -63,6 +62,7 @@
     "global-dirs": "3.0.1",
     "globby": "11.1.0",
     "has-yarn": "2.1.0",
+    "indent-string": "4.0.0",
```
Used to be engine-core dependency/

Jolg42 commented:
Note that `engine-core` package might be used externally, though I only found this public repo, still active, using it:
https://github.com/search?q=repo%3Amohammed-bahumaish%2Fprisma-editor%20engine-core&type=code
So I think it looks ok to move on.

SevInf commented on helpers/compile/plugins/fill-plugin/fillers/perf_hooks.ts:
```diff
@@ -0,0 +1,4 @@
+const performance = globalThis['performance']
```
Was not needed before because OTL packages in `engines` core had different version, which did not call `performance.now()`. Will be irrelevant again once we stop bundling OTL packages in the client.

SevInf committed:
Polyfill performance for Node 14

aqrln commented on .github/CODEOWNERS:
```diff
@@ -1,15 +1,18 @@
 # the repo. Unless a later match takes precedence,
-*                                @Jolg42 @jkomyno @prisma/client-team-ts
+
+-                                @Jolg42 @jkomyno @prisma/client-team-ts
```
These changes look unintentional (and probably breaking), looks like the file was reformatted as Markdown.

SevInf replied:
They totally are, thank you, fixed

Jolg42 commented on .github/CODEOWNERS:
```diff
@@ -1,15 +1,18 @@
 # the repo. Unless a later match takes precedence,
-*                                @Jolg42 @jkomyno @prisma/client-team-ts
+
+-                                @Jolg42 @jkomyno @prisma/client-team-ts
```
```suggestion
*                                @Jolg42 @jkomyno @prisma/client-team-ts
```

SevInf committed:
Fix CODEOWNERS

SevInf committed:
chore(client): Merge client and engines-core packages

Moves most of `engine-core` package into `core/runtime/engine`
subdirectory of the client.
Moves some utitlites from the `engine-core` into various `internals` sub
modules.

This allows us to use `internals` helpers within what used to be
`engines-core` code. In the future, it would also allow us to simplify
the code by reusing some of the client-side abstractions.

Fix prisma/client-planning#358
Unblock prisma/client-planning#340

aqrln reviewed (approved):
💯
